src/sapling/saplingscriptpubkeyman.h

```diff
@@ -106,10 +106,18 @@ class SaplingNoteData
         READWRITE(nullifier);
         READWRITE(witnesses);
         READWRITE(witnessHeight);
+        READWRITE(amount);
+        READWRITE(address);
+        READWRITE(memo);
     }
 
     friend bool operator==(const SaplingNoteData& a, const SaplingNoteData& b) {
-        return (a.ivk == b.ivk && a.nullifier == b.nullifier && a.witnessHeight == b.witnessHeight);
+        return (a.ivk == b.ivk &&
+                a.nullifier == b.nullifier &&
+                a.witnessHeight == b.witnessHeight &&
+                a.amount == b.amount &&
+                a.address == b.address &&
+                a.memo == b.memo);
```

Comment:
Do we need to add this three values to the equal operator?

Reply:
At least the address for external notes (since they have no ivk nor witnesses)

Reply:
second thoughts about this, are we comparing notes entry at all?
We possibly can remove this moving on.